src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -589,6 +590,36 @@ void *Simulator::sending_trampoline(void * /*unused*/)
 	return nullptr;
 }
 
+void Simulator::send_sim_tone()
+{
+	uint64_t time_now = hrt_absolute_time();
+
+	// Don't grab the next tone until the current one is scheduled to be done.
+	if (_tone_timer < time_now) {
+
+		struct {
+			unsigned int frequency = 0;
+			unsigned int duration = 0;
+			unsigned int silence = 0;
+			uint8_t volume = 0;
+		} tone_info = {};
+
+		if (_tunes.get_next_note(tone_info.frequency, tone_info.duration, tone_info.silence) > 0) {
```

Comment:
Don't you also have to `get` the `volume`? Otherwise `volume` is still `0` and we never send the message.

Reply:
Nice catch.  Fixed. Thanks!